src/main/scala/wdl4s/cwl/ExpressionTool.scala

```diff
@@ -0,0 +1,31 @@
+package wdl4s.cwl
+
+import shapeless.{:+:, CNil}
+import wdl4s.cwl.CwlVersion._
+
+case class ExpressionTool(
+                           inputs: Array[InputParameter] :+: Map[InputParameter#Id, InputParameter#`type`] :+: Map[InputParameter#Id, InputParameter] :+: CNil,
+                           outputs: Array[ExpressionToolOutputParameter] :+: Map[ExpressionToolOutputParameter#Id, ExpressionToolOutputParameter#`type`] :+: Map[ExpressionToolOutputParameter#Id, ExpressionToolOutputParameter] :+: CNil,
+                           `class`: String,
+                           expression: ECMAScriptExpression :+: String :+: CNil,
+                           id: Option[String],
+                           requirements: Option[Array[Requirement]],
+                           hints: Option[Array[String]], //TODO should be Any
+                           label: Option[String],
+                           doc: Option[String],
+                           cwlVersion: Option[CWLVersion]
+)
+
+case class ExpressionToolOutputParameter(
+                                          id: String,
+                                          label: Option[String],
+                                          secondaryFiles: Option[ECMAScriptExpression :+: String :+: Array[ECMAScriptExpression :+: String :+: CNil] :+: CNil], //TODO Use Either if/when it works
```

Comment:
hmm I'd probably vote for consistency if we know we're going to have situations with more than two types.